### PR TITLE
Add support for res-state endpoint

### PR DESCRIPTION
**Pull request by FRSTR**

### CODE REVIEW

**Comment by FRSTR** [comment]
@epenet , please take a look into the coverage tests error:
```
Name                                      Stmts   Miss Branch BrPart       Cover   Missing
-------------------------------------------------------------------------------------
src/renault_api/cli/renault_vehicle.py      133      1     34      1    99%   325
```
What test should be added, or maybe some configs updated?

File: src/renault_api/cli/renault_vehicle.py [hunk 1]
```diff
@@ -315,6 +316,28 @@ async def update_lock_status(
             update_status_table(status_table, key, value, unit)
 
 
+async def update_res_state(
+    vehicle: RenaultVehicle, status_table: Dict[str, Any]
+) -> None:
+    """Update status table from get_vehicle_res_state."""
+    try:
+        if not await vehicle.supports_endpoint("res-state"):
```
Comment (epenet) [inline src/renault_api/cli/renault_vehicle.py:324]: We do not have vehicles that do not support this endpoint yet, so we have to exclude this from code coverage. 
```suggestion
        if not await vehicle.supports_endpoint("res-state"):  # pragma: no cover
```

File: docs/endpoints/vehicle_data.res-state.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
+res-state
```
[the remark this answers is not in the document]
Comment (FRSTR) [reply docs/endpoints/vehicle_data.res-state.rst:1]: Updated there too

File: src/renault_api/cli/renault_vehicle.py [hunk 3]
```diff
@@ -315,6 +316,28 @@ async def update_lock_status(
...
+            return
+        response = await vehicle.get_res_state()
+    except QuotaLimitException as exc:  # pragma: no cover
+        raise click.ClickException(repr(exc)) from exc
+    except KamereonResponseException as exc:  # pragma: no cover
+        click.echo(f"res state: {exc.error_details}", err=True)
+    else:
+        items = [
+            ("RES details", response.details, None),
+            ("RES code", response.code, None),
```
Comment (epenet) [inline src/renault_api/cli/renault_vehicle.py:334]: I don't think we need to show the code in the CLI. The details is probably sufficient.
```suggestion
            ("Engine state", response.details, None),
```

(that would need adjusting in the CLI test results)

File: src/renault_api/kamereon/__init__.py [hunk 4]
```diff
@@ -32,6 +32,7 @@
     "hvac-settings": {"version": 1},
     "location": {"version": 1},
     "lock-status": {"version": 1},
+    "res-state": {"version": 1},
```
[the remark this answers is not in the document]
Comment (FRSTR) [reply src/renault_api/kamereon/__init__.py]: Resolving for now

File: src/renault_api/cli/renault_vehicle.py (hunk 3 above)
Comment (FRSTR) [reply src/renault_api/cli/renault_vehicle.py]: Done

File: src/renault_api/cli/renault_vehicle.py [hunk 5]
```diff
@@ -315,6 +316,27 @@ async def update_lock_status(
...
+            return
+        response = await vehicle.get_res_state()
+    except QuotaLimitException as exc:  # pragma: no cover
+        raise click.ClickException(repr(exc)) from exc
+    except KamereonResponseException as exc:  # pragma: no cover
+        click.echo(f"res state: {exc.error_details}", err=True)
+    else:
+        items = [
+            ("Engine state", response.details, None),            
+        ]
```
Comment (epenet) [inline src/renault_api/cli/renault_vehicle.py:334]: There are some trailing spaces that made their way in...
```suggestion
        items = [
            ("Engine state", response.details, None),
        ]
```

Comment (FRSTR) [reply src/renault_api/cli/renault_vehicle.py]: yeah, fixed in the next commit

**Review by epenet — APPROVED** [review]
Thanks @FRSTR 👍

**Pull request merged**